### PR TITLE
Adjust table row default height to 44px

### LINKED ISSUE #2553: [Enhancement] Kirby Data Table Row Height
- [ ] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
Row heights are too high for 14px font and larger data tables.

## Describe the solution you'd like
Row height is calculated by padding: 16px and line-height: 16px = 48px. Rows should be 44px to look good.

Reduce default padding to 4px 16px and set min-height to 44px. This would also allow visualy larger objects like toggle (32px height) to fit in table-rows without breaking the vertical rythm.

*min-height is already implemented in dropdown pop-overs

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by mark-drastrup**
## Which issue does this PR close?

This PR closes #2553

## What is the new behavior?
The table cells now have a padding and a height to ensure that the default height of the table row is 44px. The cell height will still grow if the content is higher.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any additional context?
In #2553 it is suggested to use `min-height`. As far as I can see `min-height` cannot be used cannot be used to table elements, and therefore I have used `height` instead.

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/src/lib/components/data-table/table/table.component.scss [hunk 1]
```diff
@@ -40,7 +40,8 @@ th {
 }
 
 td {
-  padding: utils.size('s');
+  height: 44px;
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/data-table/table/table.component.scss]: In general I dont think we should define a height for the table cells like this, as we are sacrificing responsiveness. I think the height should be defined by the sum of the block level padding and the line height, so the cell grows as expected when the text wraps into more lines.

Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/data-table/table/table.component.scss:43]: By removing this, I think we are good to go

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]

File: libs/designsystem/src/lib/components/data-table/table/table.component.scss (hunk 1 above)
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/data-table/table/table.component.scss]: Would be good to quickly investigate if we expect additional table cell heights in the future. Then we could add this to variables similar to item-heights: 
```
$item-heights: (
  m: 56px,
  s: 44px,
  xs: 32px,
) !default;
```

Or even modify the name of that one if they should be similar!

Comment (mark-drastrup) [reply libs/designsystem/src/lib/components/data-table/table/table.component.scss]: @RasmusKjeldgaard we still haven't received any response whether we expect additional table cel height in the future or not. Should we push for a response or should we add it later, if it becomes necessary? :)

Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/data-table/table/table.component.scss]: Let us extract it to a variable and get on with it! :D

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**